astroplan/tests/test_scheduling.py

```diff
@@ -21,70 +22,182 @@
                                      dec=89.26410897 * u.deg), name="Polaris")
 
 apo = Observer.at_site('apo')
-targets = [vega, rigel, polaris]
+targets = [vega, polaris, rigel]
+default_time = Time('2016-02-06 03:00:00')
+only_at_night = [AtNightConstraint()]
+
+
+def test_observing_block():
+    block = ObservingBlock(rigel, 1*u.minute, 0, configuration={'filter': 'b'})
+    assert(block.configuration['filter'] == 'b')
+    assert(block.target == rigel)
+    times_per_exposure = [1*u.minute, 4*u.minute, 15*u.minute, 5*u.minute]
+    numbers_of_exposures = [100, 4, 3, 12]
+    readout_time = 0.5*u.minute
+    for index in range(len(times_per_exposure)):
+        block = ObservingBlock.from_exposures(vega, 0, times_per_exposure[index],
+                                              numbers_of_exposures[index], readout_time)
+        assert(block.duration == numbers_of_exposures[index] *
+               (times_per_exposure[index] + readout_time))
+
+
+def test_slot():
+    start_time = default_time
+    end_time = start_time + 24 * u.hour
+    slot = Slot(start_time, end_time)
+    slots = slot.split_slot(start_time, start_time+1*u.hour)
+    assert len(slots) == 2
+    assert slots[0].end == slots[1].start
+
+
+def test_schedule():
+    start_time = default_time
+    end_time = start_time + 24 * u.hour
+    schedule = Schedule(start_time, end_time)
+    assert schedule.slots[0].start == start_time
+    assert schedule.slots[0].end == end_time
+    assert schedule.slots[0].duration == 24*u.hour
+    schedule.new_slots(0, start_time, end_time)
+    assert len(schedule.slots) == 1
+    new_slots = schedule.new_slots(0, start_time+1*u.hour, start_time+4*u.hour)
+    assert np.abs(new_slots[0].duration - 1*u.hour) < 1*u.second
+    assert np.abs(new_slots[1].duration - 3*u.hour) < 1*u.second
+    assert np.abs(new_slots[2].duration - 20*u.hour) < 1*u.second
+
+
+def test_schedule_insert_slot():
+    schedule = Schedule(default_time, default_time + 5*u.hour)
+    duration = 2*u.hour + 1*u.second
+    end_time = default_time + duration
+    block = TransitionBlock.from_duration(duration)
+    schedule.insert_slot(end_time - duration, block)
+    # even when float evaluation doesn't work, it should still schedule properly
+    assert not end_time - duration == default_time
```

Comment:
Could you label this test to indicate that it's designed to test where the ordinary float comparison would fail? Could you also test a case where the float comparison wouldn't fail, and that `insert_slot` still works?